circuit_passes/src/bucket_interpreter/env/mod.rs

```diff
@@ -286,6 +307,15 @@ impl<'a> Env<'a> {
         switch_impl_read!(self, get_vars_sort)
     }
 
+    pub fn collect_write(
+        &self,
+        dest_address_type: &AddressType,
+        idx: usize,
+        collector: &mut Writes,
+    ) {
+        switch_impl_read!(self, collect_write, dest_address_type, idx, collector)
```

Comment:
The function is named `collect_write` but it calls to the read trait?

Reply:
Ok, I'll rename the macros. The "read" -> "get" macro is used for functions that get some information from the current Env. The "write" -> "update" macros are used for functions that produce a new Env with updated information.